apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,198 @@
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      return await response.json();
+    } catch (error) {
+      clearTimeout(timeout);
+      if (
+        retries > 0 &&
+        !(error instanceof Error && error.name === "AbortError")
+      ) {
+        // Wait for a backoff period before retrying
+        await new Promise((resolve) => setTimeout(resolve, backoff));
+        return this.httpRequest(url, options, retries - 1, backoff * 2); // Exponential backoff
+      }
+      if (error instanceof Error) {
+        throw error;
+      } else {
+        throw new Error("An unknown error occurred");
+      }
+    }
+  }
+
+  /**
+   * Fetch the set of available price feeds.
+   * This endpoint can be filtered by asset type and query string.
+   * This will throw an error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param query Optional query string to filter the price feeds. If provided, the results will be filtered to all price feeds whose symbol contains the query string. Query string is case insensitive. Example : bitcoin
+   * @param filter Optional filter string to filter the price feeds. If provided, the results will be filtered by asset type. Possible values are crypto, equity, fx, metal, rates. Filter string is case insensitive. Available values : crypto, fx, equity, metals, rates
+   * @returns Array of PriceFeedMetadata objects.
+   */
+  async getPriceFeeds(
+    query?: string,
+    filter?: string
+  ): Promise<PriceFeedMetadata[]> {
+    const url = new URL(`${this.baseURL}/v2/price_feeds`);
+    if (query) {
+      url.searchParams.append("query", query);
+    }
+    if (filter) {
+      url.searchParams.append("filter", filter);
+    }
+    return await this.httpRequest<PriceFeedMetadata[]>(url.toString());
+  }
+
+  /**
+   * Fetch the latest price updates for a set of price feed IDs.
+   * This endpoint can be customized by specifying the encoding type and whether the results should also return the parsed price update.
+   * This will throw an error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param ids Array of hex-encoded price feed IDs for which updates are requested.
+   * @param encoding Optional encoding type. If true, return the price update in the encoding specified by the encoding parameter. Default is hex.
+   * @param parsed Optional boolean to specify if the parsed price update should be included in the response. Default is false.
+   * @returns Array of PriceUpdate objects containing the latest updates.
+   */
+  async getLatestPriceUpdates(
+    ids: HexString[],
+    encoding?: EncodingType,
+    parsed?: boolean
+  ): Promise<PriceUpdate[]> {
+    const url = new URL(`${this.baseURL}/v2/updates/price/latest`);
+    // Append parameters to the URL search parameters
+    ids.forEach((id) => url.searchParams.append("ids[]", id));
```

Comment:
It's preferable to use `for ... of` over `forEach`, see https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-for-each.md.  I know it feels less "functional" but in reality `forEach` isn't functional anyways since the only reason to use it is to trigger side effects.